src/Database/Driver/Sqlserver.php

```diff
@@ -187,12 +187,13 @@ public function enabled(): bool
      * Prepares a sql statement to be executed
      *
      * @param string|\Cake\Database\Query $query The query to prepare.
+     * @param array $options The driver options to set on the statement.
      * @return \Cake\Database\StatementInterface
      */
-    public function prepare($query): StatementInterface
+    public function prepare($query, array $options = []): StatementInterface
     {
         $this->connect();
-        $options = [PDO::ATTR_CURSOR => PDO::CURSOR_SCROLL];
+        $options = [PDO::ATTR_CURSOR => PDO::CURSOR_SCROLL] + $options;
```

Comment:
Don't you want the defaults added afterwards? This won't let `PDO::ATTR_CURSOR` be overridden by $options.

Reply:
Actually, the goal was to _add_ custom driver options, alongside `PDO::ATR_CURSOR`.

But what you said is true, in case someone wants to override that attribute, that won't work with the `+` operator

Reply:
I've submitted a check for _in case_ `PDO::ATTR_CURSOR` is set, just go through with the execution.